Fix to allow live system analysis of s390x kernels that have been

configured with CONFIG_RANDOMIZE_BASE=y (KASLR).  Without the patch,
the "--kaslr=<offset>" command line option is required.
(anderson@redhat.com)
---
Index: crash-7.2.7/symbols.c
===================================================================
--- crash-7.2.7.orig/symbols.c
+++ crash-7.2.7/symbols.c
@@ -618,6 +618,12 @@ kaslr_init(void)
 		st->_stext_vmlinux = UNINITIALIZED;
 	}
 
+	if (machine_type("S390X") &&  /* Linux 5.2 */
+	    (symbol_value_from_proc_kallsyms("__kaslr_offset") != BADVAL)) {
+		kt->flags2 |= (RELOC_AUTO|KASLR);
+		st->_stext_vmlinux = UNINITIALIZED;
+	}
+
 	if (QEMU_MEM_DUMP_NO_VMCOREINFO()) {
 		if (KDUMP_DUMPFILE() && kdump_kaslr_check()) {
 			kt->flags2 |= KASLR_CHECK;
